### ISSUE TITLE
Please don't remove `prepare_seq2seq_batch` in future versions

### ISSUE THREAD

**Issue by zartdinov** [huggingface/transformers#14255]
# 🚀 Feature request

I think we can just add some extra attribute like `pad_token_id`=-100 to ignore padding in the loss

## Motivation
It can be easily and cleanly used with `datasets` library now:

```python
encoded_train_dataset = train_dataset.map(
    lambda batch: tokenizer.prepare_seq2seq_batch(
        batch['text'], batch['summary'], padding='max_length', truncation=True, max_length=256, max_target_length=64
    ),
    batched=True,
    remove_columns=train_dataset.column_names,
)
```
Instead of writing every time custom preprocess function like here:
https://github.com/huggingface/transformers/blob/9adff7a0f49f88a6cc718a1d30088988dc78bb6a/examples/pytorch/translation/run_translation.py#L407

@sgugger @patrickvonplaten @patil-suraj
